### PR TITLE
[AiLab] Delete controller method, route and tests for ML models 

### PR DESCRIPTION
**Pull request by Erin007**
Setting up the controller action and route to delete trained models from the UserMlModel table and S3, including a couple of tests.

### CODE REVIEW

File: dashboard/app/controllers/api/v1/ml_models_controller.rb [hunk 1]
```diff
@@ -49,6 +49,14 @@ def get_trained_model
     render json: model
   end
 
+  # DELETE api/v1/ml_models/:model_id
+  def destroy
+    @user_ml_model = UserMlModel.find_by(model_id: params[:model_id])
+    return head :forbidden unless @user_ml_model.user_id == current_user.id
+    @user_ml_model.destroy
+    delete_from_s3(@user_ml_model.model_id)
+  end
```
Comment (breville) [inline dashboard/app/controllers/api/v1/ml_models_controller.rb:63]: Do we need to specify anything about what we return as content here?   `dont_cache`, `no_content`, etc.?

File: dashboard/app/controllers/api/v1/ml_models_controller.rb [hunk 2]
```diff
@@ -62,4 +70,8 @@ def upload_to_s3(model_id, trained_model)
   def download_from_s3(model_id)
     AWS::S3.download_from_bucket(S3_BUCKET, model_id)
   end
+
+  def delete_from_s3(model_id)
+    AWS::S3.delete_from_bucket(S3_BUCKET, model_id)
+  end
```
Comment (made-line) [inline dashboard/app/controllers/api/v1/ml_models_controller.rb:81]: Great work! This is new to me. Would we ever need to verify/test that this action is successful in S3?

Comment (Erin007) [reply dashboard/app/controllers/api/v1/ml_models_controller.rb:81]: [This delete method](https://github.com/code-dot-org/code-dot-org/blob/99238d5e702366b7cebf6a470a3080f589fa6782/lib/cdo/aws/s3.rb#L155) will return a boolean indicating whether the deletion worked or not. I took your idea and use it in the check above. Thanks!

File: dashboard/config/routes.rb [hunk 3]
```diff
@@ -740,10 +740,15 @@
       get 'peer_review_submissions/index', to: 'peer_review_submissions#index'
       get 'peer_review_submissions/report_csv', to: 'peer_review_submissions#report_csv'
 
-      post 'ml_models/save', to: 'ml_models#save'
-      get 'ml_models/names', to: 'ml_models#user_ml_model_names'
-      get 'ml_models/:model_id', to: 'ml_models#get_trained_model'
-      get 'ml_models/:model_id/metadata', to: 'ml_models#user_ml_model_metadata'
+      resources :ml_models, only: [:show, :destroy] do
```
Comment (Erin007) [inline dashboard/config/routes.rb:743]: @madelynkasula You suggested I use resource for the ml_models routes on an older PR - just now getting to it.  Is this what you were thinking?

Comment (maddiedierker) [reply dashboard/config/routes.rb:743]: yes, this is exactly what i was thinking! thank you for following up on that

File: dashboard/app/controllers/api/v1/ml_models_controller.rb [hunk 4]
```diff
@@ -28,27 +28,40 @@ def save
...
+  def destroy
+    @user_ml_model = UserMlModel.find_by(model_id: params[:id])
+    return head :not_found unless @user_ml_model
+    return head :forbidden unless @user_ml_model.user_id == current_user.id
+    @user_ml_model.destroy
+    deleted_from_s3 = delete_from_s3(@user_ml_model.model_id)
+    status =
+      @user_ml_model.destroyed? && deleted_from_s3 ? "success" : "failure"
+    render json: {id: @user_ml_model.id, status: status}
+    head :no_content
```
Comment (breville) [inline dashboard/app/controllers/api/v1/ml_models_controller.rb:62]: These appear to conflict with other, but so long as https://github.com/code-dot-org/code-dot-org/pull/39924 takes care of it, then I guess that's okay.

**Review by breville — APPROVED** [review]

**Pull request merged**